.github/workflows/preview-on-pr-update.yml

```diff
@@ -0,0 +1,69 @@
+name: preview_on_pr_update
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  sonarqube:
+    name: analyze
+    # only run when updating an 'Open' PR
+    if: github.event.pull_request.state == 'open'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Disabling shallow clone is recommended for improving relevancy of reporting
+          fetch-depth: 0
+      - name: sonarqube-scan-pullrequest
+        uses: sonarsource/sonarqube-scan-action@master
+        if: ${{ github.base_ref == 'main' }}
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        with:
+          args: >
+            -Dsonar.pullrequest.key=${{ github.event.number }}
+            -Dsonar.pullrequest.branch=${{ github.event.pull_request.head.ref }}
+            -Dsonar.pullrequest.base=${{ github.base_ref }}
+            -Dsonar.qualitygate.wait=true
+            -Dsonar.qualitygate.timeout=1000
+      
+      - name: sonarqube-scan-branch
+        uses: sonarsource/sonarqube-scan-action@master
+        if: ${{ github.base_ref != 'main' }}
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        with:
+          args: >
+            -Dsonar.branch.name=${{ github.event.pull_request.head.ref }}
```

Comment:
@codVenturer - Think you only need the `sonarqube-scan-pullrequest` job which runs the quality gate analysis against the pull request.

This one is for running the analyzer on entire branch. This should be in a different workflow which runs on whenever changes are pushed to `main` branch.

Reply:
Hey, so for this I am creating, another yml file for branch and put this code there. it should be named `production-on-push` right?

Reply:
**`production_on_push`**

Yes, you'd have to create a new workflow